File: cases/taint/cpp/object-separation-negative/object_separation_negative.cpp
struct Holder {
    int value;
};

int dfb_source() { // DFB-SOURCE: object-separation-input
    return 1;
}

void dfb_sink(int value) {} // DFB-SINK: object-separation-sink

void run() {
    Holder tainted;
    Holder clean;
    tainted.value = dfb_source(); // DFB-WITNESS: object-separation-store
    clean.value = 0;
    dfb_sink(clean.value);
}

File: cases/taint/php/object-separation-negative/object_separation_negative.php
<?php
class Holder
{
    public string $value = "clean";
}

function dfb_source(): string { // DFB-SOURCE: object-separation-input
    return "tainted";
}

function dfb_sink(string $value): void {} // DFB-SINK: object-separation-sink

function run(): void {
    $tainted = new Holder();
    $clean = new Holder();
    $tainted->value = dfb_source(); // DFB-WITNESS: object-separation-store
    $clean->value = "clean";
    dfb_sink($clean->value);
}
